### ISSUE TITLE
How to get CP and AD to work together(Question)

### ISSUE THREAD

**Issue by alpha117** [Courseplay/courseplay#4071]
Morning Guys
I've tried and failed!

CP does the field work then when it wants refilling how does it link to AD for the refill course?

What do I need to do to make them work together

Thanks

**Comment by Opa-Andre** [comment]
As you say refilling, I assume you want to fertilize / lime your field… 
First make sure you're using the actual versions of the CP & AD mods.

You already should have created an AD network created with defined named network points to your field and to your refilling station. In AD select Mode 5 (loading course - its the next one after "combine unload"). In the upper line you select your stop point near the field, in the lower line you select your loading station (for example fertilizer sellpoint or silo) and depending of your used equipment what you want to refill (fertilizer, lime, seeds, etc.). Now start running the AD course. If your vehicle attachments (seeder, fertilizer etc.) are currently completely empty, it will drive to the refill Station, refill and then drive to your field Destination. If it's not empty, it will drive directly to your field. 

When arriving at the field destination, AD will wait - so this time you'll have to switch off AD.

Now use CP to create your field work route. Before starting the course, select in the CP Options menu  that you want to use AD for refilling. 

Now start the CP fieldwork course.  

What will happen:
CP is doing the fieldwork. As soon as your implements are empty, CP will "internally" make a note of the actual fieldwork position and will switch on AD automatically. AD will then check for the closest AD waypoint in its driving network and steer directly to it. As it reaches the network it then will drive its own route from it to the refill Station, refill and drive back to the field destination. Then it will signal that it reached "the end" of its course. CP will then take over again, driving straight from this point to the before noted fieldwork position and continue with the fieldwork.  

Two notes: 

1. Make sure in your AD driving network that you are able to reach the network from any place / location in the field without obstacles being in your way. The same of course applies for the way from the fieldborder (AD field destination point) to any location within the field.

2. Plan your setup on beforehand because when you run into problems (for example obstacles in the way during the time while CP is using AD for refilling) then stop CP first, before stopping AD.  Otherwise if you just stop AD manually, CP will interpret this as "refill done" and will use a straight way from the current position of your vehicle back to the current field work points.

**Comment by alpha117** [comment]
Many thanks for that detailed reply. Off to have a go on Fenton Forest 4X (field2)

**Comment by alpha117** [comment]
just completed field #2 and it works PERFECTLY...wow  what a combination CP + AD

great work guys

Many thanks

**Comment by pvaiko** [comment]
Check out 274, AD should do a better job while driving on the field.

**Comment by alpha117** [comment]
Hi Pvaiko,
Sure will do. 
Many thanks

**Comment by Opa-Andre** [comment]
@pvaiko Cool optimization, shows the positive will of the AD & CP creators to work together for an even better user experience! Thanks much for that - to all of you!

**Comment by alpha117** [comment]
> @pvaiko Cool optimization, shows the positive will of the AD & CP creators to work together for an even better user experience! Thanks much for that - to all of you!

totally agree, fantastic work to all parties

**Comment by alpha117** [comment]
Could this be achieved?

1. AD used to refill then drive to field.....this work great already
2. IF CP has a 'saved' field course for that implement then AD automatically stops and CP starts the field course

Thanks

**Comment by Tensuko** [comment]
Should already work, but don't ask me how to set up AD for this, I am still playing arround with it myself...

**Comment by Tensuko** [comment]
The 2nd option is not working, sry I misunderstood that.
So you mean you just tell AD to drive to destination X and it will load the Field course ?
I doubt that is possible, as you would have to assign a Course to a tool.

Or more like, you load a CP Course and then use AD to send it to the Field and CP Starts working on the 1st waypoint ?

**Comment by Tensuko** [comment]
Closed due to inactivity.
if got any idea, make new Issue and we will talk about that there and I mark it as Suggestion.
